packages/runtime/src/utils/__tests__/partition.test.ts

```diff
@@ -0,0 +1,47 @@
+import { expectTypeOf } from 'expect-type'
+
+import { partition, partitionRecord } from '../partition'
+
+function isNumber(value: unknown): value is number {
+  return typeof value === 'number'
+}
+
+describe('partition', () => {
+  it('splits the array into two based on a predicate', () => {
+    const array = [1, 2, 3, 4, 5]
+    const [even, odd] = partition(array, x => x % 2 === 0)
+    expect(even).toStrictEqual([2, 4])
+    expect(odd).toStrictEqual([1, 3, 5])
+  })
```

Comment:
I'd consider adding a test to make sure the method returns an empty partition when none of the values match the predicate. I know the function signature protects us from someone accidentally modifying it to return `null` or `undefined`, but I feel that it's a helpful addition.

Reply:
Good idea.